### DISCUSSION TITLE
Async not used

### DISCUSSION THREAD

**Discussion by naidu-rohit** [fastapi/full-stack-fastapi-template#2120] [Questions] [answered]
The template does not use async at all. Is there any particular reason for that? Do celery, sqlalchemy etc work with async? 
How much code will have to be changed for async?

**Comment by KrunchMuffin** [comment]
I was trying to find out that info. So no asyncpg?

**Comment by theonesud** [comment]
I couldn't find a way to use sqlalchemy with async-await for my previous project so I decided to use the [Databases package](https://github.com/encode/orm)

**Comment by littleforce163** [comment]
For database async application, tortoise-orm is fine too.

**Comment by jamesgalt-paminnov** [comment]
sqlalchemy 1.4 enables async use of the ORM

**Comment by gerazenobi** [comment]
I had similar/related doubts (why the examples were using blocking operations) and I was missing some background information that I found reading [this comment](https://github.com/tiangolo/fastapi/issues/260#issuecomment-495945630) and the technical details in the [fastapi documentation](https://fastapi.tiangolo.com/async/#very-technical-details) 

But particularly for your question you should find your answer here:
- [this comment](https://github.com/tiangolo/fastapi/issues/2947#issuecomment-798792950)
- [this other comment](https://github.com/tiangolo/full-stack-fastapi-postgresql/issues/174#issuecomment-640051483)
- [this PR](https://github.com/tiangolo/fastapi/pull/2331)

**Comment by tiangolo** [comment, marked as answer]
The simplest is to start without async, and add async to the code paths that need it, but that requires careful handling to make sure the libraries used internally are compatible with async, or are properly wrapped with, for example, Asyncer.
